ui_framework/components/tabs/tabs.test.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import { render, shallow } from 'enzyme';
+import { requiredProps } from '../../test/required_props';
+import sinon from 'sinon';
+
+import {
+  KuiTabs,
+} from './tabs';
+
+import {
+  KuiTab,
+} from './tab';
+
+describe('KuiTabs', () => {
+  const tabs = [
+    'Cobalt',
+    'Dextrose',
+    'Helium-3',
+    'Monosodium Glutamate',
+  ];
+
+  describe('throws an error', () => {
```

Comment:
Intuitively, I would say: If they can be written for re-use, a library of prop-types sounds like a good idea. If they are absolutely component-specific (which should be pretty rare, I can't think of such a case right now) they should live with the component (maybe in a local `prop_types.js`).

Stubbing `console` to test them feels really crude, since they are supposed to be pure functions. But if there are other instances of this, then I would agree that they could be cleanup up later in one go.